File: strategy-pattern/py/context.py
from strategy_interface import StrategyInterface


class Context:

    def __init__(self, model: StrategyInterface) -> None:
        self._model = model

    def switch_model(self, model: StrategyInterface):
        self._model = model

    def process_input(self):
        return self._model.generateResponse()

    

File: strategy-pattern/ts/js/context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var AIContext = /** @class */ (function () {
    function AIContext(model) {
        this.model = model;
    }
    AIContext.prototype.switchModel = function (model) {
        this.model = model;
    };
    AIContext.prototype.processInput = function () {
        return this.model.generateResponse();
    };
    return AIContext;
}());
exports.default = AIContext;
